src/phoenix/utilities/ratelimits.py

```diff
@@ -0,0 +1,210 @@
+import asyncio
+import sys
+import time
+from collections import defaultdict
+from functools import wraps
+from typing import Any, Awaitable, Callable, Dict, Optional, TypeVar, Union, cast
+
+if sys.version_info < (3, 10):
+    from typing_extensions import ParamSpec
+else:
+    from typing import ParamSpec
```

Comment:
conditionals like this doesn't feel like it serves us any good - unless there are runtime considerations which I don't think it does?

Reply:
I think sometimes the implementations do differ, but in this specific case it looks like the `typing_extensions` module already does all of this logic